src/Intercom/Clients/UsersClient.cs

```diff
@@ -335,21 +335,43 @@ public User IncrementUserSession(String id)
             return result.Result;
         }
 
+        public User IncrementUserSession(String id, List<String> companyIds)
+        {
+            if (String.IsNullOrEmpty(id))
+            {
+                throw new ArgumentNullException(nameof(id));
+            }
+
+            if (companyIds == null)
+            {
+                throw new ArgumentNullException(nameof(companyIds));
+            }
+
+            if (!companyIds.Any())
+            {
+                throw new ArgumentException("'companyIds' shouldnt be empty.");
+            }
+
+            ClientResponse<User> result = null;
+            String body = JsonConvert.SerializeObject(new { id = id, new_session = true, companies = companyIds.Select(c => new { id = c }) });
+            result = Post<User>(body);
+            return result.Result;
+        }
+
         public User IncrementUserSession(User user)
         {
             if (user == null)
             {
                 throw new ArgumentNullException(nameof(user));
             }
 
-            if (String.IsNullOrEmpty(user.id))
+            if (String.IsNullOrEmpty(user.id) && String.IsNullOrEmpty(user.user_id) && string.IsNullOrEmpty(user.email))
             {
-                throw new ArgumentException("'user.id' argument is null.");
+                throw new ArgumentException("you need to provide either 'user.id', 'user.user_id', 'user.email' to increase a user session.");
             }
 
             ClientResponse<User> result = null;
-            String body = JsonConvert.SerializeObject(new { id = user.id, new_session = true });
-            result = Post<User>(body);
+            result = Post<User>(Transform(user));
```

Comment:
By passing a Transform we allow the user to have a `new_session` attribute and take that automatically as well as other details. Before, even though we were passing a `User` we were only adding the `id` and passing a `new_session` without companies. This also allows updating the session for all those companies in the user object.